### PR TITLE
Update links for 0.2 release of icerpc-csharp.

### LINKED ISSUE #382: Broken links to examples
The recent reorganization of the examples in icerpc-csharp broke many links to examples from docs.icerpc.dev, since all these links reference main. They should reference the latest release branch, currently 0.1.x:

https://github.com/icerpc/icerpc-csharp/tree/0.1.x

### PR DESCRIPTION
**Pull request by ReeceHumphreys**
This PR updates the example links to point to their new locations. Notably, I did change the link that was previously from GreeterCore (No IDL) to just `slice/Greeter` since we seem to have gotten rid of GreeterCore here:
https://github.com/icerpc/icerpc-csharp/commit/4d51a79bc0aeec76a98558fa9d62f8a581a4fd84#diff-71d356b15c94c747b9ea8c2f332efd9075afe085d324e76978c4b4c186252630

**Edit:**
Missed the intent of the issue, this PR really updated the csharp-0.2-dev docs to point to moved examples. Not a priotity to merge this. As this will only be needed for when we release the  csharp-0.2-dev docs build.

### CODE REVIEW

File: content/slice/language-guide/parameters.md [hunk 1]
```diff
@@ -76,11 +81,12 @@ case. The type of the C# field is the mapped C# type for `Type`.
 Tagged parameters are mapped just like regular parameters. The tag and tag number don't appear in the mapped C# API.
 
 {% slice2 %}
-### Stream parameters in C#
+### Stream parameters in C #
```
Comment (pepone) [inline content/slice/language-guide/parameters.md:84]: This doesn't look correct, the linter seems to get confused when a title ends with `#`

**Review by pepone — APPROVED** [review]

File: components/tags/examples.tsx [hunk 2]
```diff
@@ -10,17 +10,18 @@ export const Examples = () => {
       <ExampleCard
         title="GreeterLog"
         description="A typical IceRPC application."
-        href="https://github.com/icerpc/icerpc-csharp/tree/0.1.x/examples/GreeterLog"
+        href="https://github.com/icerpc/icerpc-csharp/tree/main/examples/slice/Logger"
```
Comment (bernardnormier) [inline components/tags/examples.tsx:13]: The title needs to be resynchronized as well.

**Review by bernardnormier — APPROVED** [review]
We need a more thorough update of these cards. Now that Slice and Protobuf are on equal footing, it's odd to link to the Slice Logger example without mentioning Slice or that there is an equivalent Protobuf example.

**Comment by ReeceHumphreys** [comment]
> We need a more thorough update of these cards. Now that Slice and Protobuf are on equal footing, it's odd to link to the Slice Logger example without mentioning Slice or that there is an equivalent Protobuf example.

Yes, I agree - I think we will want to rethink how many cards we have and what we show.

**Pull request merged**